common/src/app/common/types/tokens_lib.cljc

```diff
@@ -1287,6 +1311,17 @@ Will return a value that matches this schema:
           (assoc-in ["$metadata" "activeThemes"] active-themes-clear)
           (assoc-in ["$metadata" "activeSets"] active-sets))))
 
+  (decode-single-set-json [this set-name tokens]
```

Comment:
This function does not requires access to anything internal of the data structure, so it should be implemented using a simple function and not a protocol method.

Reply:
Also, the name indicates decode, but it also transforms the current tokensLib.
I think this should be defined on first instance as decode function  and then use already available methods on token lib for adding the decoded values.
This function is only used on tokens-dictionary ns, so inclusive, the helper that does both decode+add can be implemented there and not in the data structure.

Reply:
@not-in-stock those changes are important, but we have agreed on doing them in a separate branch, for the next release.

Reply:
done in https://github.com/penpot/penpot/pull/6221